src/AppInstallerCLICore/PortableInstaller.cpp

```diff
@@ -0,0 +1,496 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "ExecutionContext.h"
+#include "PortableInstaller.h"
+#include "winget/Manifest.h"
+#include "winget/ManifestCommon.h"
+#include "winget/Filesystem.h"
+#include "winget/PathVariable.h"
+#include "Microsoft/PortableIndex.h"
+#include "Microsoft/Schema/IPortableIndex.h"
+#include <AppInstallerErrors.h>
+
+using namespace AppInstaller::Utility;
+using namespace AppInstaller::Registry;
+using namespace AppInstaller::Registry::Portable;
+using namespace AppInstaller::Registry::Environment;
+using namespace AppInstaller::Repository;
+using namespace AppInstaller::Repository::SQLite;
+using namespace AppInstaller::Repository::Microsoft;
+using namespace AppInstaller::Repository::Microsoft::Schema;
+
+namespace AppInstaller::CLI::Portable
+{
+    namespace
+    {
+        constexpr std::wstring_view c_PortableIndexFileName = L"portable.db";
+
+        bool VerifyPortableExeHash(const std::filesystem::path& targetPath, const std::string& hashValue)

```

Comment:
It seems like there should already be a VerifyHash function somewhere for validating other downloaded installers. Is this duplicate code? If not, perhaps this should be moved to Common as `ValidateFileHash` or something of that nature, since I doubt it applies just to portable EXEs